backends/backend_libkcapi.c

```diff
@@ -0,0 +1,2365 @@
+        slen = data->S.len;
+
+        /*
+        Metadata for complete key = 0x30 and sum of length of all fields
+
+        BER encoding the length of any field
+        actual length of a field = 0x02 and len
+
+        if length <= 127 - 1 byte (actual length)
+        if length > 127 and length <= 255 - 2 bytes (Byte1 = 0x81, Byte2 = actual length)
+        if length > 255 - 3 bytes (Byte1 = 0x82, Byte2 and Byte3 = actual length)
+        */
+
+
+        total = rlen;
+        if(rlen <= 127)
+                total = total + 2;
+        else if(rlen >= 128 && rlen <= 255)
+                total = total + 3;
+        else
+                total = total + 4;
+
+        total = total + slen;
+        if(slen <= 127)
+                total = total + 2;
+        else if(slen >= 128 && slen <= 255)
+                total = total + 3;
+        else
+                total = total + 4;
+
+        if(total <= 127)
+                extra = 2;
+        else if(total >= 128 && total <= 255)
+                extra = 3;
+        else
+                extra = 4;
+
+        CKINT_LOG(alloc_buf(total + extra + data->msg.len + 1, &in), "ecdsa: in buffer could not be allocated\n");
+        ptr = (&in)->buf;
+        ptr[0] = 0x30;
+        if(extra == 2)
+                memcpy(ptr + 1, &total, 1);
+        if(extra == 3)
+        {
+                ptr[1] = 0x81;
+                memcpy(ptr + 2, &total, 1);
+        }
+        if(extra == 4)
+        {
+                tmp =(unsigned char*) (&total);
+                ptr[1] = 0x82;
+                memcpy(ptr + 2, tmp + 1, 1);
+                memcpy(ptr + 3, tmp , 1);
+        }
+
+        ptr = ptr + extra;
+        ptr = write_field(ptr, data->R.buf, rlen);
+        ptr = write_field(ptr, data->S.buf, slen);
+
+        if(data->msg.buf)
+                memcpy(ptr + 1, data->msg.buf, data->msg.len);
+
+        ret = kcapi_akcipher_verify(handle, in.buf, in.len, dgst.buf, dgst.len, 0);
+
+        if(ret < 0)
+        {
+                data->sigver_success = 0;
+                logger(LOGGER_ERR, "ecdsa: SigVer Failed\n");
+        }
+        else
+        {
+                data->sigver_success = 1;
+                logger(LOGGER_DEBUG, "ecdsa: SigVer Success\n");
+        }
+        ret = 0;
+
+out:
+        kcapi_akcipher_destroy(handle);
+        if(in.buf)
+                free_buf(&in);
+        if(dgst.buf)
+                free_buf(&dgst);
+        if(pk.buf)
+                free_buf(&pk);
+        return ret;
+}
+
+static struct ecdsa_backend kcapi_ecdsa =
+{
+        NULL,
+        ecdsa_keygen,
+        ecdsa_keyver,
+        NULL,
+        ecdsa_sigver,
+        NULL,
+        NULL
+};
+ACVP_DEFINE_CONSTRUCTOR(kcapi_ecdsa_backend)
+static void kcapi_ecdsa_backend(void)
+{
+        register_ecdsa_impl(&kcapi_ecdsa);
+}
+
+static int gcm_encrypt(struct aead_data *data, flags_t parsed_flags)
+{
+        struct kcapi_handle *handle = NULL;
+        char *cipher = "gcm(aes)";
```

Comment:
As discussed, please use existing function to return the name, instead of hardcoding.